### PR TITLE
NEXT-00000 - allow exporting invalid records with import only profiles

### PR DESCRIPTION
**Pull request by jeboehm**
### 1. Why is this change necessary?
When using import only profiles it is not possible to see invalid records of a failed import.

### 2. What does this change do, exactly?
ImportExportService uses the same (import-only) profile to generate and export the invalid records, even on console.
The check that ensures that an import-only profile is only used for imports is now skipped when the activity is exporting invalid records.

### 3. Describe each step to reproduce the issue or behaviour.
- Create a random import profile
- Disallow exports
- Set some required fields
- Upload a csv without these missing fields
- The import will fail, it is not possible to export the failed records, users are getting a error message like "The import/export profile with id 018d7e9fc06b76f7bf09ecbcb9364146 can only be used for import"

### 5. Checklist

- [x] I have rebased my changes to remove merge conflicts
- [x] I have written tests and verified that they fail without my change
- [x] I have created a [changelog file](https://github.com/shopware/platform/blob/trunk/adr/2020-08-03-implement-new-changelog.md) with all necessary information about my changes
- [ ] I have written or adjusted the documentation according to my changes
- [x] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfil them.

### CODE REVIEW

**Comment by AydinHassan** [comment]
hey @jeboehm is it finished already? It's still labelled as a WIP :)

**Comment by jeboehm** [comment]
Hi @AydinHassan 
Yes, PR is finished. :)

File: src/Core/Content/ImportExport/Service/ImportExportService.php [hunk 1]
```diff
@@ -59,7 +59,8 @@ public function prepareExport(
     ): ImportExportLogEntity {
         $profileEntity = $this->findProfile($context, $profileId);
 
-        if (!\in_array($profileEntity->getType(), [ImportExportProfileEntity::TYPE_EXPORT, ImportExportProfileEntity::TYPE_IMPORT_EXPORT], true)) {
+        if (!\in_array($profileEntity->getType(), [ImportExportProfileEntity::TYPE_EXPORT, ImportExportProfileEntity::TYPE_IMPORT_EXPORT], true)
```
Comment (larskemper) [inline src/Core/Content/ImportExport/Service/ImportExportService.php:62]: Please add a test for your changes

File: changelog/_unreleased/2024-02-07-import-only-profile-edge-case.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
+---
+title: Import / Export - Import only profile edge case
+issue: NEXT-00000
+author_github: @jeboehm
+---
+
```
Comment (larskemper) [inline changelog/_unreleased/2024-02-07-import-only-profile-edge-case.md]: Please remove this empty line

**Review by larskemper — CHANGES_REQUESTED** [review]

**Comment by jeboehm** [comment]
Check! ✅

File: src/Core/Content/Test/ImportExport/Service/ImportExportServiceTest.php [hunk 3]
```diff
@@ -241,6 +242,26 @@ public function testExportProfileShouldThrowExceptionInImport($profile, $task, $
         }
     }
 
+    #[DataProvider('profileProvider')]
+    public function testExportProfileShouldNotThrowExceptionWhenExportingErrorLog($profile, $task): void
```
Comment (larskemper) [inline src/Core/Content/Test/ImportExport/Service/ImportExportServiceTest.php:246]: Good job :muscle:,
but it would be better if you extend the `profileProvider()` with your test case(s) and just use the `shouldThrowException` parameter to test a "should not throw..." scenario.
Additionally, it might be beneficial to rename the initial test, `testExportProfileShouldThrowExceptionInImport()`, to better align with the "should" and "should not" test cases.

Comment (jeboehm) [reply src/Core/Content/Test/ImportExport/Service/ImportExportServiceTest.php:246]: Hi @LarsKemper 
sorry, I don't get it.

Since the error log export must not throw an exception, what do you want me to do with the $shouldThrowException parameter?
What name do you suggest for testExportProfileShouldThrowExceptionInImport?

Comment (larskemper) [reply src/Core/Content/Test/ImportExport/Service/ImportExportServiceTest.php:246]: Hey :wave:, 
sorry for the late reply. What I mean is simply that you unify both test to one, and you mirror the test cases via the props. That way, we only use the provider once and reduce the number of runs. In other words, if there is no exception thrown: `shouldThrowException = false`...
As a name, I would perhaps use something like `testProfileExportAndImportPreparation`. So something that fits better for both cases. But of course it is a matter of opinion (`testExportProfileShouldThrowExceptionInImport` is also ok :wink:).

Comment (jeboehm) [reply src/Core/Content/Test/ImportExport/Service/ImportExportServiceTest.php:246]: I don't like the idea. Your suggestion would combine three different test cases into one, and the name "testProfileExportAndImportPreparation" would cover none of them. 
I'd guess that it's okay to leave these cases separated, even performance-wise. The test only contains a few profiles.

**Comment by mitelg** [comment]
thanks for your contribution @jeboehm 👍🎉💙

**Comment by jeboehm** [comment]
You are very welcome! :)
